src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,45 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_SIZE = 10;
+    private int currentFilling = 0;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[STORAGE_SIZE];
+        values = (V[]) new Object[STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < currentFilling; i++) {
+            if ((keys[i] == key) || (keys[i] != null && keys[i].equals(key))) {
+                values[i] = value;
+                return;
+            }
+        }
+        if (currentFilling != STORAGE_SIZE) {
+            values[currentFilling] = value;
+            keys[currentFilling] = key;
+            currentFilling++;
+        } else {
+            throw new RuntimeException("Current storage is full");
+        }
```

Comment:
as an optimal alternative
```suggestion
        if (size >= STORAGE_SIZE) {
            throw new RuntimeException("Current storage is full");
        }
         values[currentFilling] = value;
         keys[currentFilling] = key;
         currentFilling++;
```